### ISSUE TITLE
Suggestion and autocompletion for all string properties

### ISSUE THREAD

**Issue by falsandtru** [microsoft/TypeScript#12138]
It would be very helpful to add suggestion-based accessibility for properties that cannot access using dots.

```ts
interface Alternative {
  '<|>'(): void;
}
declare var a: Alternative;
a. // suggest `<|>` property.
a['<|>'] // autocompletion selects a best style per property name.
```

**Comment by RyanCavanaugh** [comment]
This doesn't come up terribly often and there are some architectural issues with certain editors that make this insertion impractical, unfortunately
